docs/src/tutorials/software-management.md

```diff
@@ -39,34 +40,51 @@ Find more information about [managing the device software](https://cumulocity.co
 
 1. In the Software repository page, click Add software at the right of the top menu bar.
 2. In the resulting dialog box,
-   - to add a new software, enter a name for the software (and confirm it by clicking Create new in the resulting window),
+   * to add a new software, enter a name for the software (and confirm it by clicking Create new in the resulting window),
      a description and its version.
-   - to add a new version, select the software for which you want to add a new version from the dropdown list in the Software
+   * to add a new version, select the software for which you want to add a new version from the dropdown list in the Software
      field and enter a version of the package. The version is optional and can be left with a white space, meaning the latest version.
    > Note: The version field format is `package_version::plugin_type_name`.The plugin type name that is provided here is used to pick
      the appropriate plugin among those installed in /etc/tedge/sm-plugins.
+   > Note 2: If the postfix `::plugin_type_name` is left empty a default plugin will be used if defined, otherwise `apt` will be used.
 3. Optionally, you can define the device type filter when adding a new software.
-4. As of now thin-edge.io only supports software from apt repositories. So, select the **Provide a file path** option and give an **empty space**.
-5. Click Save.
+4. thin-edge.io ships a default plugin supporting `debian` packages from both `apt` repositories as well as remote locations. Figure 1.
+   If you prefer to use packages from an `apt` repository, select the **Provide a file path** option and give an **empty space** (' ').
+   If you would like to use other sources (eg. file uploaded to your cloud or an external source), provide the full url to the file. Figure 2.
+   If you would like to use your cloud storage select `Upload a binary` option and upload the file to your cloud. Figure 3.
+
+   > Note: Bear in mind that some external remotes may require additional authentication which is not supported at the moment.
+5. Press `Add Software` button.
 
 Follow above mentioned steps as shown in the diagram
 
  ![Add new software](./images/add-new-software-to-repo.png)
 
+Figure 2.
+ ![Add new software from uploaded binary](./images/add-new-software-to-repo-binary.png)
```

Comment:
Can these images be added inline to the bullet points where the descriptions are? Otherwise the rendered doc with 3 images back to back looks weird.

On a side note, the formatting of the new images is totally different from that of Figure 1. The new ones look better as the original one was appearing stretched. If you could replace Fig.1 also with the same styling, it would have been better (if it's not too much trouble).

Reply:
Sure, good suggestion, I'll also update the original image.